components/SaveTemplate.vue

```diff
@@ -45,19 +42,26 @@ export default {
     }
   },
   methods: {
-    save(event) {
-      const currentState = this.$store.state
+    ...mapState({
+      state: state => state,
+    }),
+    async save() {
+      const currentState = this.state
       console.log('uploading current state', currentState)
-      uploadTemplate(this.title, this.description, this.filename, currentState)
-        .then((id) => {
-          console.log('Successfully uploaded template', id)
-        })
-        .catch((error) => {
-          console.log('ERROR uploading template', error)
-        })
+      try {
+        const id = await uploadTemplate(
+          this.title,
+          this.description,
+          this.filename,
+          currentState
+        )
+        console.log('Successfully uploaded template', id)
+      } catch (error) {
+        console.error('ERROR uploading template', error)
```

Comment:
If you're leaving all of these in for your future self that's cool but I'll flag them as I go along :)

Reply:
Theres some of them that are just useful logging and some that are just mess I forgot to delete